core/src/test/java/org/elasticsearch/gateway/RecoveryFromGatewayIT.java

```diff
@@ -332,48 +332,39 @@ public void testLatestVersionLoaded() throws Exception {
         String metaDataUuid = client().admin().cluster().prepareState().execute().get().getState().getMetaData().clusterUUID();
         assertThat(metaDataUuid, not(equalTo("_na_")));
 
-        Map<String, long[]> primaryTerms = assertAndCapturePrimaryTerms(null);
-
         logger.info("--> closing first node, and indexing more data to the second node");
-        internalCluster().fullRestart(new RestartCallback() {
+        internalCluster().stopRandomDataNode();
 
-            @Override
-            public void doAfterNodes(int numNodes, Client client) throws Exception {
-                if (numNodes == 1) {
-                    logger.info("--> one node is closed - start indexing data into the second one");
-                    client.prepareIndex("test", "type1", "3").setSource(jsonBuilder().startObject().field("field", "value3").endObject()).execute().actionGet();
-                    // TODO: remove once refresh doesn't fail immediately if there a master block:
-                    // https://github.com/elastic/elasticsearch/issues/9997
-                    client.admin().cluster().prepareHealth("test").setWaitForYellowStatus().get();
-                    client.admin().indices().prepareRefresh().execute().actionGet();
-
-                    for (int i = 0; i < 10; i++) {
-                        assertHitCount(client.prepareSearch().setSize(0).setQuery(matchAllQuery()).execute().actionGet(), 3);
-                    }
+        logger.info("--> one node is closed - start indexing data into the second one");
```

Comment:
the test starts two nodes, shut down one and while it's down makes changes to the other. The it shuts the other one down and brings both of the up together. I had some trouble with the full restart in the early versions of the change and I decided to simplify the test to make it cleared what it does